RecipeApp/AuthorizationViewController/AuthorizationViewController.swift

```diff
@@ -0,0 +1,466 @@
+            emailTextField.layer.borderWidth = 0.0
+            warningEmailLabel.isHidden = true
+        }
+    }
+
+    /// Обновление визуала для пароля на основе результата валидации
+    func updateValidationPassword(result: String) {
+        if result == "isEmpty" {
+            passwordTextField.layer.borderColor = UIColor.clear.cgColor
+            passwordTextField.layer.borderWidth = 0.0
+            passwordLabel.textColor = UIColor(named: "loginColor")
+            warningPassLabel.isHidden = true
+        } else if result == "noPass" {
+            passwordLabel.textColor = .red
+            passwordTextField.layer.borderColor = UIColor.red.cgColor
+            passwordTextField.layer.borderWidth = 1.0
+            passwordTextField.layer.cornerRadius = 8
+            warningPassLabel.isHidden = false
+        } else {
+            passwordLabel.textColor = UIColor(named: "loginColor")
+            passwordTextField.layer.borderColor = UIColor.clear.cgColor
+            passwordTextField.layer.borderWidth = 0.0
+            warningPassLabel.isHidden = true
+        }
+    }
+
+    /// Обработчик нажатия кнопки видимости пароля
+    @objc private func togglePasswordVisibility() {
+        presenter?.createPasswordVisibilityButton(tapped: passwordTextField.isSecureTextEntry.self)
+    }
+
+    /// метод отработки скрытие пароля для презентора
+    func updatePasswordVisibilityButton(result: Bool) {
+        if result == true {
+            passwordTextField.isSecureTextEntry.toggle()
+        } else {
+            passwordTextField.isSecureTextEntry.toggle()
+        }
+    }
+
+    /// обработчик нажатия кнопки логин
+    @objc private func loginButtonTapped() {
+//        presenter?.openVC()
+        /// скрытие текста кнопки
+        loginButton.setTitle("", for: .normal)
+
+        /// создание и настройка индикатора загрузки
+        let activityIndicator = UIActivityIndicatorView(style: .medium)
+        activityIndicator.color = .white
+        activityIndicator.center = CGPoint(x: loginButton.bounds.midX, y: loginButton.bounds.midY)
+        activityIndicator.startAnimating()
+        loginButton.addSubview(activityIndicator)
+
+        /// таймер на 3 секунды
+        Timer.scheduledTimer(withTimeInterval: 3.0, repeats: false) { _ in
+            /// остановка анимации и скрытие ее
+            activityIndicator.stopAnimating()
+            activityIndicator.removeFromSuperview()
+
+            self.presenter?.loginButtonTapped(
+                login: self.emailTextField.text ?? "",
+                password: self.passwordTextField.text ?? ""
+            )
+        }
+    }
+
+    /// метод отработки появление плашки для презентора
+    func updateLoginButton(result: Bool) {
+        if result == false {
+            errorView.isHidden = false
+            errorMessageLabel.isHidden = false
+        } else {
+            errorView.isHidden = false
+            errorMessageLabel.isHidden = false
+        }
```

Comment:
```suggestion
            errorView.isHidden = false
            errorMessageLabel.isHidden = false
```
на обоих условиях одно и тоже
проверьте что там у вас